src/main/java/Parser.java

```diff
@@ -0,0 +1,106 @@
+// what command should occur based on the what the user input
+// makes sense of the text
+
+public class Parser {
+    static String exitInput = "bye";
+
+    static Command parse(String text) {
+
+        if (text.toLowerCase().equals(exitInput)) {
+            return new ByeCommand();
+        }
+
+        if (text.toLowerCase().equals("list")) {
+            return new ListCommand();
+        }
+
+        String[] splitStr = text.split("\\s+");
+
+        if (splitStr[0].toLowerCase().equals("done")) {
+            int index = Integer.parseInt(splitStr[1]) - 1;
+            return new DoneCommand(index);
+        }
+
+        if (splitStr[0].toLowerCase().equals("delete")) {
+            int index = Integer.parseInt(splitStr[1]) - 1;
+            return new DeleteCommand(index);
+        }
+
+        if (splitStr[0].toLowerCase().equals("deadline")) {
+            try {
+                LocalDate date = LocalDate.now().minus(1, ChronoUnit.MONTHS);
+                String deadline = "";
+                for (int i = 1; i < splitStr.length; i++) {
+                    if ((splitStr[i].equals("/by"))) {
+                        date = LocalDate.parse(splitStr[i + 1]);
+                        break;
+                    } else {
+                        deadline += splitStr[i] + " ";
+                    }
+                }
+                deadline = deadline.substring(0, deadline.length() - 1);
+
+                if (date.isBefore(LocalDate.now())) {
+                    throw new DukeException("☹ OOPS!!! You cannot set a date that is in the past! Format is yyyy-mm-dd ☹ OOPS!!!");
+                } else {
+                    Deadline d = new Deadline(deadline, date);
+                    return new DeadlineCommand(d);
+                }
+            } catch (DukeException e) {
+                System.out.println(e);
+            }
+        }
+
+        if (splitStr[0].toLowerCase().equals("event")) {
+            try {
+                String at = "";
+                String event = "";
+                for (int i = 1; i < splitStr.length; i++) {
+                    if ((splitStr[i].equals("/at"))) {
+                        for (int j = i + 1; j < splitStr.length; j++) {
+                            at += splitStr[j] + " ";
+                        }
+                        break;
+                    } else {
+                        event += splitStr[i] + " ";
+                    }
+                }
+                if (at.equals("")) {
+                    throw new DukeException("☹ OOPS!!! Where is this event????? use /at to tell me! ☹ OOPS!!!");
+                } else {
+                    event = event.substring(0, event.length() - 1);
+                    at = at.substring(0, at.length() - 1);
+
+                    Event e = new Event(event, at);
+                    return new EventCommand(e);
+                }
+            } catch (DukeException error) {
+                System.out.println(error);
```

Comment:
Would it be better if you output the errors in the output error stream using System.err.println()?